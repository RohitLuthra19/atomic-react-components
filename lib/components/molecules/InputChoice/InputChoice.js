// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import { InputVanilla as Input } from '../../atoms/Input';
import Label from '../../atoms/Label';
import styles from './InputChoice.style';

type Props = {
  id: string,
  type: string,
  disabled?: boolean,
  checked?: boolean,
  className?: string,
  children?: Node,
  label?: string,
  labelProps?: Object,
  showLabelBeforeInput?: boolean,
};

const InputChoice = ({
  id,
  type,
  checked,
  disabled,
  className,
  label,
  children,
  showLabelBeforeInput,
  labelProps,
  ...otherProps
}: Props): Node => (
  <Label
    className={className}
    htmlFor={id}
    showLabelBeforeInput={showLabelBeforeInput}
    {...labelProps}
  >
    <Input id={id} type={type} defaultChecked={checked} disabled={disabled} {...otherProps} />
    <span type={type} />
    <div>{children || label}</div>
  </Label>
);

InputChoice.defaultProps = {
  checked: undefined,
  disabled: false,
  className: '',
  children: '',
  label: '',
  labelProps: {},
  showLabelBeforeInput: false,
};

export default styled(InputChoice)`
  ${styles};
`;
export { InputChoice as InputChoiceVanilla };
