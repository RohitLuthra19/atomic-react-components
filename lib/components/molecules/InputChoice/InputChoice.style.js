import { css } from 'styled-components';

const styles = css`
  cursor: ${props => (props.disabled ? 'default' : 'pointer')};
  display: inline-flex;
  > input {
    display: none;
  }
  > span {
    border: 2px solid
      ${props => (props.disabled ? props.theme.colors.CTA_DISABLED : props.theme.TEXT_INVERSE)};
    width: 20px;
    height: 20px;
    position: relative;
    display: inline-block;
    vertical-align: middle;

    ${props =>
      props.type === 'radio' &&
      `
      border-radius: 50%;
    `} &::after {
      content: '';
      width: 12px;
      height: 12px;
      left: 50%;
      top: 50%;
      transform: translate(-50%, -50%) rotate(-45deg);
      position: absolute;

      ${props =>
        props.type === 'checkbox' &&
        `
        height: 6px;
        border: 2px solid #fff;
        border-width: 0 0 2px 2px;
        top: calc(50% - 3px);
      `} ${props =>
        props.type === 'radio' &&
        `
        border-radius: 50%;
      `};
    }
  }
  & > input:checked + span {
    ${props =>
      props.type === 'radio'
        ? `border-color: ${
            props.disabled ? props.theme.colors.CTA_DISABLED : props.theme.TEXT_INVERSE
          };`
        : ''} ${props =>
      props.type === 'checkbox' &&
      `
      border-color: ${props.disabled ? props.theme.colors.CTA_DISABLED : props.theme.TEXT_INVERSE};
      background-color: ${
        props.disabled ? props.theme.colors.CTA_DISABLED : props.theme.TEXT_INVERSE
      };
      border-radius: 2px;
    `}

    &:after {
      ${props =>
        props.type === 'radio' &&
        `
        background-color: ${
          props.disabled ? props.theme.colors.CTA_DISABLED : props.theme.TEXT_INVERSE
        };
      `};
    }
  }

  ${props =>
    props.showLabelBeforeInput
      ? `
    display: flex;
    justify-content: space-between;
    & > span {
      order: 1;
    }
  `
      : `
    & > span {
      margin-right: 5px;
    }
  `};
`;

export default styles;
