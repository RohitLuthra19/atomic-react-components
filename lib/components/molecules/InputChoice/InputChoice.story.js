/* eslint-disable no-unused-vars */
// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import { checkbox, inputRadio } from './InputChoice.mock';
import InputChoice, { InputChoiceVanilla } from './InputChoice';

storiesOf('Molecules', module)
  .addParameters({ jest: ['InputChoice', 'InputChoiceVanilla'] })
  .add('InputChoice Knobs', () => <InputChoiceVanilla {...checkbox} />)
  .add('InputChoice Checkbox', () => <InputChoice {...checkbox} />)
  .add('InputChoice Radio', () => <InputChoice {...inputRadio} />);
