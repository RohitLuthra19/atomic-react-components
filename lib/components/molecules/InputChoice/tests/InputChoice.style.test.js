import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import InputChoice from '../InputChoice';
import Theme from '../../../../styles/themes/base';

describe('InputChoice', () => {
  let inputControl = '';
  const theme = { ...Theme };

  test('should render radio correctly', () => {
    inputControl = shallow(<InputChoice theme={theme} type="radio" id="test_1" name="test_1" />);
    expect(inputControl).toMatchSnapshot();
  });

  test('should render checkbox correctly', () => {
    inputControl = shallow(<InputChoice theme={theme} type="checkbox" id="test_2" name="test_2" />);
    expect(inputControl).toMatchSnapshot();
  });
});
