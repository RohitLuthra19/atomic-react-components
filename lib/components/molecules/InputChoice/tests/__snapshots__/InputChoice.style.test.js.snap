// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputChoice should render checkbox correctly 1`] = `
.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 > input {
  display: none;
}

.c0 > span {
  border: 2px solid #006193;
  width: 20px;
  height: 20px;
  position: relative;
  display: inline-block;
  vertical-align: middle;
}

.c0 > span::after {
  content: '';
  width: 12px;
  height: 12px;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) rotate(-45deg);
  -ms-transform: translate(-50%,-50%) rotate(-45deg);
  transform: translate(-50%,-50%) rotate(-45deg);
  position: absolute;
  height: 6px;
  border: 2px solid #fff;
  border-width: 0 0 2px 2px;
  top: calc(50% - 3px);
}

.c0 > input:checked + span {
  border-color: #006193;
  background-color: #006193;
  border-radius: 2px;
}

.c0 > span {
  margin-right: 5px;
}

<InputChoice
  checked={false}
  className="c0"
  disabled={false}
  id="test_2"
  label=""
  labelProps={Object {}}
  name="test_2"
  showLabelBeforeInput={false}
  theme={
    Object {
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "borderColorBoldest": "#c00",
      "borderColorSecondary": "#d9d9d9",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "fieldBackgroundColor": "#ffffff",
      "fieldBackgroundDisabledColor": "#f3f3f3",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fieldTextColor": "#222",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "fontSizeHighlight": "18px",
      "letterSpacing": "0px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "linkColorSecondary": "#016fad",
      "maxWidthPageContent": "80rem",
      "textColorBrandPrimary": "#006193",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSecondaryLight": "#666",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zLoader": 1000,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zSticky": 700,
      },
    }
  }
  type="checkbox"
/>
`;

exports[`InputChoice should render radio correctly 1`] = `
.c0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 > input {
  display: none;
}

.c0 > span {
  border: 2px solid #006193;
  width: 20px;
  height: 20px;
  position: relative;
  display: inline-block;
  vertical-align: middle;
  border-radius: 50%;
}

.c0 > span::after {
  content: '';
  width: 12px;
  height: 12px;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) rotate(-45deg);
  -ms-transform: translate(-50%,-50%) rotate(-45deg);
  transform: translate(-50%,-50%) rotate(-45deg);
  position: absolute;
  border-radius: 50%;
}

.c0 > input:checked + span {
  border-color: #006193;
}

.c0 > input:checked + span:after {
  background-color: #006193;
}

.c0 > span {
  margin-right: 5px;
}

<InputChoice
  checked={false}
  className="c0"
  disabled={false}
  id="test_1"
  label=""
  labelProps={Object {}}
  name="test_1"
  showLabelBeforeInput={false}
  theme={
    Object {
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "borderColorBoldest": "#c00",
      "borderColorSecondary": "#d9d9d9",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "fieldBackgroundColor": "#ffffff",
      "fieldBackgroundDisabledColor": "#f3f3f3",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fieldTextColor": "#222",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "fontSizeHighlight": "18px",
      "letterSpacing": "0px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "linkColorSecondary": "#016fad",
      "maxWidthPageContent": "80rem",
      "textColorBrandPrimary": "#006193",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSecondaryLight": "#666",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zLoader": 1000,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zSticky": 700,
      },
    }
  }
  type="radio"
/>
`;
